### PR TITLE
Use with blocks to write to temp files

### PR DESCRIPTION
**Pull request by lucc**
Like #922

### CODE REVIEW

File: alot/db/utils.py [hunk 1]
```diff
@@ -314,13 +314,12 @@ def extract_body(mail, types=None, field_key='copiousoutput'):
...
-                                                          prefix=prefix,
-                                                          suffix=suffix)
-                    # write payload to tmpfile
-                    tmpfile.write(raw_payload)
-                    tmpfile.close()
-                    tempfile_name = tmpfile.name
+                    with tempfile.NamedTemporaryFile(
+                            delete=False, prefix=prefix, suffix=suffix) \
+                            as tmpfile:
+                        # write payload to tmpfile
```
Comment (lucc) [inline alot/db/utils.py]: I think this comment is also not needed. Opinions?

File: alot/commands/envelope.py [hunk 2]
```diff
@@ -375,12 +375,13 @@ def openEnvelopeFromTmpfile():
...
-                                                            suffix='.eml')
-        self.envelope.tmpfile.write(content.encode('utf-8'))
-        self.envelope.tmpfile.flush()
-        self.envelope.tmpfile.close()
+        with tempfile.NamedTemporaryFile(delete=False, prefix='alot.',
+                                         suffix='.eml') as tmpfile:
+            tmpfile.write(content.encode('utf-8'))
+            # Is this needed if the file is closed at the end of the with
+            # block?
+            tmpfile.flush()
```
Comment (lucc) [inline alot/commands/envelope.py]: Does one of you know if that is needed?

File: alot/db/utils.py (hunk 1 above)
Comment (dcbaker) [inline alot/db/utils.py]: it seems pretty obvious to me what's going on here, I don't think we need the comment.

File: alot/commands/thread.py [hunk 3]
```diff
@@ -944,13 +944,10 @@ def apply(self, ui):
...
 
-                tmpfile = tempfile.NamedTemporaryFile(delete=False,
-                                                      prefix=prefix,
-                                                      suffix=suffix)
-
-                tempfile_name = tmpfile.name
-                self.attachment.write(tmpfile)
-                tmpfile.close()
+                with tempfile.NamedTemporaryFile(delete=False, prefix=prefix,
+                                                 suffix=suffix) as tmpfile:
```
Comment (dcbaker) [inline alot/commands/thread.py]: You're wrapped this and the one after in the opposite manner, from a consistancy point of view I'd prefer to see one or the other.

File: alot/commands/envelope.py (hunk 2 above)
Comment (dcbaker) [inline alot/commands/envelope.py]: I would leave it, flush is basically a blocking operation to ensure that the file moves out of the harddrive cache and is actually written to disk (which is basically a  no-op on SSDs and NVMe).

**Comment by lucc** [comment]
@dcbaker I amended with your comments and force pushed.

**Comment by pazz** [comment]
thanks guys

**Pull request merged**